### PR TITLE
Add doublesTier

### PR DESCRIPTION
**Pull request by KrisXV**
This is the server-side part of adding `doublesTier`

### CODE REVIEW

File: sim/dex.js [hunk 1]
```diff
@@ -853,8 +859,10 @@ class ModdedDex {
 			case 'pokemontag':
 				// valid pokemontags
 				const validTags = [
-					// pokemon tiers
+					// singles tiers
 					'uber', 'ou', 'bl', 'uu', 'bl2', 'ru', 'bl3', 'nu', 'bl4', 'pu', 'nfe', 'lcuber', 'lc', 'cap', 'caplc', 'capnfe',
+					// doubles tiers
+					'duber', 'dou', 'dbl', 'duu', 'dnfe',
```
Comment (Zarel) [inline sim/dex.js]: >dnfe

no

Comment (Zarel) [reply sim/dex.js]: "NFE" is not a tier, it's a pokemon tag that doesn't need a doubles version.

**Comment by KrisOnline** [comment]
fixed @Zarel

**Comment by Zarel** [comment]
I would not add `doublesTier` to the dex for NFE and LC pokemon at all.

**Comment by KrisOnline** [comment]
done again @zarel

**Comment by TheImmortal** [comment]
Why is it "DOU", "DUber", etc... The tier is OU, Ubers, etc. just like singles. You're already classifying it as doubles with ``doublesTier``. If you were doing something like ``tiers: ['OU', 'DOU']`` where you combined singles and doubles then I can understand. But as it stands, I say you should name it the same as singles' tiers.

**Comment by Zarel** [comment]
Mostly because searching etc is clearly with a D in front of it.

**Comment by Zarel** [comment]
iow: namespace collisions are serious, the D helps a lot with that.

File: chat-plugins/datasearch.js [hunk 2]
```diff
@@ -122,6 +122,7 @@ exports.commands = {
 		`Search categories are: type, tier, color, moves, ability, gen, resists, recovery, priority, stat, weight, height, egg group.`,
 		`Valid colors are: green, red, blue, white, brown, yellow, purple, pink, gray and black.`,
 		`Valid tiers are: Uber/OU/BL/UU/BL2/RU/BL3/NU/BL4/PU/NFE/LC/CAP/CAP NFE/CAP LC.`,
+		`Valid doubles tiers are: DUber/DOU/DBL/DUU/DNFE.`,
```
Comment (Zarel) [inline chat-plugins/datasearch.js]: DNFE is not in fact a valid doubles tier

Comment (KrisOnline) [reply chat-plugins/datasearch.js]: I know, but thats how you search for mons that appear as nfe in doubles

Comment (Zarel) [reply chat-plugins/datasearch.js]: What does "appear as NFE in doubles" mean? All Pokémon that are NFE in singles are NFE in doubles...?

**Comment by Zarel** [comment]
bump?

File: chat-plugins/datasearch.js (hunk 2 above)
Comment (KrisOnline) [reply chat-plugins/datasearch.js]: The description is technically "NFEs not in a higher tier", and the NFEs in higher tiers are different in singles than in doubles. For example, Porygon2 is DUU in Doubles and RU in Singles, making it an NFE in a higher tier for both formats. However, Golbat is NU in Singles but NFE in Doubles, meaning that it falls under "NFEs not in a higher tier" for Doubles but not Singles.

**Comment by KrisOnline** [comment]
ok i did it

File: chat-plugins/datasearch.js (hunk 2 above)
Comment (Zarel) [reply chat-plugins/datasearch.js]: I don't like this definition of "NFE" and I don't consider it useful, and it also contracts what "NFE" actually stands for.

"NFE" as a pokemon tag should include all NFEs, whether or not their `tier === 'NFE'`. Just like "LC" as a pokemon tag should include all pokemon legal in LC, whether or not their `tier === 'LC'`.

Comment (KrisOnline) [reply chat-plugins/datasearch.js]: removed

**Comment by KrisOnline** [comment]
what the fuck

**Comment by KrisOnline** [comment]
WHERE IS EVERYTHING

**Comment by Zarel** [comment]
It looks like you deleted your branch?

**Comment by Zarel** [comment]
Maybe you botched a rebase... Look for the old code in your reflog.

**Comment by KrisOnline** [comment]
i fixed IT

**Comment by KrisOnline** [comment]
ok everything is fixed and updated

**Comment by Zarel** [comment]
I should merge this, but then I'd need to rewrite the client table generators...

**Comment by KrisOnline** [comment]
>I should merge this, but then I'd need to rewrite the client table generators...

I don't see why this is an issue, considering everything that this adds has no direct effect on the client. Also, the client table generators will probably need to be adjusted soon anyway because of the whole BL renaming issue in IS.

**Comment by Zarel** [comment]
I know.

File: chat-plugins/info.js [hunk 3]
```diff
@@ -426,8 +426,8 @@ exports.commands = {
 					}
 					return this.sendReply(buffer);
 				case 'pokemon':
-					let template = Object.assign({}, mod.getTemplate(newTarget.name));
-					if ((room && room.id === 'smogondoubles') || (room && ['gen7doublesou', 'gen7doublesubers', 'gen7doublesuu'].includes(toId(Dex.getFormat(room.battle.name))))) template.tier = template.doublesTier;
+					let template = mod.getTemplate(newTarget.name);
+					if (room.id === 'smogondoubles' || ['gen7doublesou', 'gen7doublesubers', 'gen7doublesuu'].includes(Dex.getFormat(room.battle).id)) template.tier = template.doublesTier;
```
Comment (Zarel) [inline chat-plugins/info.js]: no :|

Comment (Zarel) [reply chat-plugins/info.js]: just pass an override tier to `getDataPokemonHTML`

Comment (KrisXV) [reply chat-plugins/info.js]: im confused by what you mean

Comment (Zarel) [reply chat-plugins/info.js]: You're overwriting `template`, which is a big no-no

Comment (KrisXV) [reply chat-plugins/info.js]: I understand that but how would I handle it in getDataPokemonHTML I’ve tried before

Comment (Zarel) [reply chat-plugins/info.js]: You could just, not do it and leave it to me, I guess.

Comment (Zarel) [reply chat-plugins/info.js]: Actually I'll just show you the code:

```
Chat.getDataPokemonHTML = function (template, gen = 7, tierString = '') {
```

...

```
	buf += '<span class="col numcol">' + (tierString || template.tier) + '</span> ';
```

Comment (KrisXV) [reply chat-plugins/info.js]: I didn't notice that you'd pushed this already so I've been spending the past hour working on this and wondering why my PR wasn't updating

**Pull request merged**